cmd/deploy/deploy.go

```diff
@@ -348,3 +373,136 @@ func getProxyHandler(name, token string, clusterConfig *rest.Config) (http.Handl
 	return handler, nil
 
 }
+
+func (dc *deployCommand) waitUntilDeployed(ctx context.Context, opts *Options) error {
+	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, opts.Name)
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(opts.Timeout)
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("pipeline '%s' didn't finish after %s", opts.Name, opts.Timeout.String())
+		case <-ticker.C:
+			isReady := true
+			dList, err := deployments.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, dep := range dList {
+				if dep.Status.ReadyReplicas != dep.Status.Replicas {
+					isReady = false
+				}
+			}
+			sfsList, err := statefulsets.List(ctx, okteto.Context().Namespace, labelSelector, c)
+			if err != nil {
+				return err
+			}
+			for _, sfs := range sfsList {
+				if sfs.Status.ReadyReplicas != sfs.Status.Replicas {
+					isReady = false
+				}
+			}
+
+			if !isReady {
+				continue
+			}
+			return nil
+		}
+	}
+}
+
+func (dc *deployCommand) showEndpoints(ctx context.Context, opts *Options, spinner *utils.Spinner) error {
+	spinner.Update("retrieving endpoints")
+	pods, err := getDeployPodNames(ctx, opts)
+	if err != nil {
+		return err
+	}
+
+	c, _, err := okteto.GetK8sClient()
+	if err != nil {
+		return err
+	}
+	endpoints, err := endpoints.GetEndpointsByPodName(ctx, pods, okteto.Context().Namespace, c)
+	if err != nil {
+		return err
+	}
+
+	svcsName := []string{}
```

Comment:
If you create a map[string]bool with service names, it would simplify the check in `ingresses.isPathBackendInSvcList` saving a nested loop